doc/lua_api.md

```diff
@@ -6753,9 +6753,10 @@ Formspec
 * `core.show_formspec(playername, formname, formspec)`
     * `playername`: name of player to show formspec
     * `formname`: name passed to `on_player_receive_fields` callbacks.
-      It should follow the `"modname:<whatever>"` naming convention.
-    * `formname` must not be empty, unless you want to reshow
-      the inventory formspec without updating it for future opens.
+        * It should follow the `"modname:<whatever>"` naming convention.
+        * If empty: Reshows the inventory formspec. Servers and
+          clients >= 5.12.0 will update the inventory formspec
+          (`ObjectRef:set_inventory_formspec`) for future opens.
```

Comment:
so isn't this equivalent to the following?
```lua
player:set_inventory_formspec(fs)
core.show_formspec(player:get_player_name(), "", fs)
```
what's the point then?

Reply:
Yes it is. It's mainly for consistency such that the server only works with the most recently sent formspec.

Reply:
The point of this PR is to allow future set_inventory_formspec calls to update the formspec if it is shown by show_formspec. So you can now use `show_formspec` to open the inventory formspec, and set_inventory_formspec will continue to work

Reply:
So currently doing `core.show_formspec(player:get_player_name(), "", fs)` where `fs ~= player:get_inventory_formspec()` causes some inconsistency/brokenness?

Reply:
>  So currently doing [...] causes some inconsistency/brokenness?

Yes. The client needs to know whether the player inventory is open such that newly sent values via `inventory_formspec` are respected (done by `PlayerInventoryFormSource`). `handleCommand_InventoryFormSpec` does not open the inventory but may update it.